db/addons.go

```diff
@@ -56,3 +56,110 @@ func (d *Database) ListAddons(ctx context.Context, opts ...QueryOption) ([]Addon
 
 	return addons, nil
 }
+
+func (d *Database) ToggleAddonPaid(ctx context.Context, addonID string, opts ...QueryOption) (*Addon, error) {
+	tx, err := d.Begin()
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	opts = append(opts, WithTX(tx))
+
+	_, db := d.querySettings(opts...)
+
+	ds1 := db.Update(t.Addons).
+		Set(goqu.Record{"default_paid": goqu.L("NOT default_paid")}).
+		Where(t.Addons.Col("id").Eq(addonID)).
+		Executor()
+
+	_, err = ds1.ExecContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	ds2 := db.From(t.Addons).
+		Select(
+			t.Addons.Col("id"),
+			t.Addons.Col("name"),
+			t.Addons.Col("description"),
+			t.Addons.Col("default_amount"),
+			t.Addons.Col("default_paid"),
+
+			t.ResourceTypes.Col("id").As(goqu.C("resource_types.id")),
+			t.ResourceTypes.Col("name").As(goqu.C("resource_types.name")),
+			t.ResourceTypes.Col("unit").As(goqu.C("resource_types.unit")),
+		).
+		Join(t.ResourceTypes, goqu.On(t.Addons.Col("resource_type_id").Eq(t.ResourceTypes.Col("id")))).
+		Where(t.Addons.Col("id").Eq(addonID)).
+		Executor()
```

Comment:
I don't know if this is annoying to do with `goqu` or otherwise non-preferred, but it would possibly be nicer database-wise to do this as one query instead of two. I think for how this is structured that'd be by a query like:

```sql
WITH new_addons AS (UPDATE addons SET ... RETURNING id, name, description, default_amount, default_paid)
  SELECT ... FROM new_addons JOIN ...
```

This would mean only one time looking through the addons table rather than two. Probably not a big deal if it's a pain to change though, again.

Reply:
Shouldn't be too bad to try out at least. I was struggling to figure out how to get goqu to do that and gave up, but I can try again.